File: executor.py
import subprocess


def run_command(command: str) -> str:
    """Run a CLI command string and return stdout/stderr."""
    try:
        result = subprocess.run(
            command,
            shell=True,
            capture_output=True,
            text=True,
            timeout=15,
        )
        return (result.stdout or result.stderr or "done").strip()
    except subprocess.TimeoutExpired:
        return "timed out"
    except Exception as e:
        return f"error: {e}"

File: src/executor.ts
import { exec } from 'child_process';
import { promisify } from 'util';

const execAsync = promisify(exec);

export async function runCommand(command: string): Promise<string> {
    try {
        const { stdout, stderr } = await execAsync(command, { timeout: 15000 });
        return (stdout || stderr || 'done').trim();
    } catch (error: any) {
        if (error.killed) {
            return 'timed out';
        }
        return `error: ${error.message}`;
    }
}
